### PR TITLE
Update cli help text for bal persist

### PR DESCRIPTION
**Pull request by dinukaamarasinghe817**
## Purpose
To update the cli help text for `bal persist`

Fixes : https://github.com/ballerina-platform/ballerina-lang/issues/42541

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 1]
```diff
@@ -5,20 +5,18 @@ SYNOPSIS
...
-       For example:
+       directories and configuration files. For example:
 
-           $ bal persist init
+           $ bal persist add
 
-       Defining the application's data model is the next step. Use the generated
-       definition file in the `persist` directory to define the data model
+       Define the application's data model as the next step. Use the generated
+       definition file in the "persist" directory to define the data model
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:17]:
```suggestion
       definition file in the `persist` directory to define the data model
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 2]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
                Department department;
            |};
 
-       Then, generate the client API and SQL script based on the data model. Use the generated
-       API to query and manipulate the persistent data in the application and use the generated
-       SQL script to create the database tables.
+       Then, generate the client API based on the data model. You can use the `bal build` command to generate the client.
+       This process will automatically regenerate the client whenever you modify the data model during project builds.
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:35]:
```suggestion
       This process regenerates the client when modifying the data model during project builds.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 3]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
+
+       Apart from the above commands, the following commands are also available:
+
+       $ bal persist init
+
+       This command initializes the package for persistence.
+       This command will create a new directory named 'persist' in the project directory. The 'persist' directory will
+       contain empty model definition file which can be used to define the data model of the Ballerina project.
+
+       $ bal persist generate --module persist --datastore mysql
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:49]:
```suggestion
       $ bal persist generate --module persist --datastore db
```

Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/42542/commits/10e8107d9a0e35db389b4d748ba1b8effc819468

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 4]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
+
+       Apart from the above commands, the following commands are also available:
+
+       $ bal persist init
+
+       This command initializes the package for persistence.
+       This command will create a new directory named 'persist' in the project directory. The 'persist' directory will
+       contain empty model definition file which can be used to define the data model of the Ballerina project.
+
+       $ bal persist generate --module persist --datastore db
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:49]:
```suggestion
       $ bal persist generate --module persist --datastore mysql
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help (hunk 2 above)
Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/42542/commits/10e8107d9a0e35db389b4d748ba1b8effc819468

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help (hunk 1 above)
Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/42542/commits/10e8107d9a0e35db389b4d748ba1b8effc819468

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 5]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
 
 BALLERINA COMMANDS
-       The available subcommands are listed below.
+       The below is a list of available subcommands:
 
-       init            Initialize the package for persistence
-       generate        Generate the client API and SQL script based on the data model defined in the `persist` directory
+       add             Initialize the package for persistence and integrate the client generation to
+                       the `bal build` command.
+       init            Initialize the package for persistence and create the "persist" directory and data model file.
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:63]:
```suggestion
       init            Initialize the package for persistence and create the `persist` directory and data model file.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 6]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
+       generate        Generate the client API based on the data model defined in the "persist" directory
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:64]:
```suggestion
       generate        Generate the client API based on the data model defined in the `persist` directory
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 7]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
                Department department;
            |};
 
-       Then, generate the client API and SQL script based on the data model. Use the generated
-       API to query and manipulate the persistent data in the application and use the generated
-       SQL script to create the database tables.
+       Then, generate the client API based on the data model. You can use the `bal build` command to generate the client.
+       This process regenerates the client when modifying the data model during project builds.
+       Use the generated API to query and manipulate the persistent data in the application.
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:36]:
```suggestion
       Then, generate the client API based on the data model. Use the `bal build` command to generate the client API.
       During project builds, this process updates the client API whenever modifications are made to the data model.
       Use the generated API to query and manipulate the persistent data in the application.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 8]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
+
+       $ bal persist init
+
+       This command initializes the package for persistence.
+       This command will create a new directory named 'persist' in the project directory. The 'persist' directory will
+       contain empty model definition file which can be used to define the data model of the Ballerina project.
+
+       $ bal persist generate --module persist --datastore mysql
+
+       This command generates the client API based on the data model defined in the "persist" directory. Usually used
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:51]:
```suggestion
       This command generates the client API based on the data model defined in the 'persist' directory. Usually used
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 9]
```diff
@@ -5,20 +5,18 @@ SYNOPSIS
...
 DESCRIPTION
-       Manage data persistence in a Ballerina package.
+       Manage the data persistence in a Ballerina package.
 
-       Initializing the package for persistence is the first step towards achieving
+       Initializing the package for persistence is the first step toward achieving
        data persistence in a Ballerina package. This step creates the required
-       directories and configuration files.
-       For example:
+       directories and configuration files. For example:
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:12]:
```suggestion
       directories and updates the configuration files to integrate the generation of client API into the package build process. For example:
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 10]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
+       $ bal persist init
+
+       This command initializes the package for persistence.
+       This command will create a new directory named 'persist' in the project directory. The 'persist' directory will
+       contain empty model definition file which can be used to define the data model of the Ballerina project.
+
+       $ bal persist generate --module persist --datastore mysql
+
+       This command generates the client API based on the data model defined in the "persist" directory. Usually used
+       for one time generation of the client.
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:52]:
```suggestion
       for one-time generation of the client API.
```

Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/commit/ef73dd6fdbd23ed185882b2a23767ffaa7c67725

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help (hunk 9 above)
Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/commit/ef73dd6fdbd23ed185882b2a23767ffaa7c67725

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help (hunk 8 above)
Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/commit/ef73dd6fdbd23ed185882b2a23767ffaa7c67725

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 11]
```diff
@@ -33,21 +31,38 @@ DESCRIPTION
...
-       Then, generate the client API and SQL script based on the data model. Use the generated
-       API to query and manipulate the persistent data in the application and use the generated
-       SQL script to create the database tables.
+       Then, generate the client API based on the data model. You can use the `bal build` command to generate the client.
+       This process regenerates the client when modifying the data model during project builds.
+       Use the generated API to query and manipulate the persistent data in the application.
        For example:
 
-       $ bal persist generate
+       $ bal build
```
[the remark this answers is not in the document]
Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Sure, I've already removed it in https://github.com/ballerina-platform/ballerina-lang/commit/ef73dd6fdbd23ed185882b2a23767ffaa7c67725

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help (hunk 7 above)
Comment (dinukaamarasinghe817) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/commit/ef73dd6fdbd23ed185882b2a23767ffaa7c67725

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 12]
```diff
@@ -33,21 +32,40 @@ DESCRIPTION
...
-       generate        Generate the client API and SQL script based on the data model defined in the `persist` directory
+       add             Initialize the package for persistence and integrate the 
+                       client generation to the 'bal build' command.
+       init            Initialize the package for persistence and create the 
+                       'persist' directory and data model file.
+       generate        Generate the client API based on the data model defined 
+                       in the 'persist' directory
+       pull            Introspect the existing database schema and generate data model
+       migrate         Generate migration scripts to update the database based 
+                       on the Ballerina schema changes
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:69]:
```suggestion
                       on the data model changes
```

**Pull request closed without merging**